node/src/service/nonce_monitor.rs

```diff
@@ -33,12 +35,18 @@ impl From<jsonrpsee::core::Error> for Error {
 	}
 }
 
-async fn rpc_request(handlers: &RpcHandlers, request: &str) -> Result<serde_json::Value, Error> {
+async fn rpc_request<T: DeserializeOwned>(
+	handlers: &RpcHandlers,
+	request: &str,
+) -> Result<Option<T>, Error> {
 	let (response, _stream) = handlers.rpc_query(request).await?;
 
-	let result = serde_json::from_str(&response).map_err(Error::Serde)?;
+	log::trace!(target: "nonce-monitor", "Got response: {:?}", response);
+
+	let result: RpcResponse<'_, Option<T>> =
+		serde_json::from_str(&response).map_err(Error::Serde)?;
 
-	Ok(result)
+	Ok(result.result)
```

Comment:
@nathanwhit do we have some sort of a test/assertion on this? We should be able to figure out when such things change under the hood !

Reply:
FTR with or without this change metrics report:

```
authority_offchain_nonce{chain="local_testnet"} 0
authority_onchain_nonce{chain="local_testnet"} 0
```

I don't see anything different so it's hard to figure out what's the change in behavior.

Reply:
Tests in https://github.com/gluwa/creditcoin/pull/1347

passing in https://github.com/gluwa/creditcoin/actions/runs/6278268305/job/17052055832?pr=1347